torch/testing/_internal/inputgen/variable/type.py

```diff
@@ -0,0 +1,70 @@
+import math
+from enum import Enum
+
+import torch
+
+
+class ScalarDtype(Enum):
+    bool = bool
+    int = int
+    float = float
+
+
+SUPPORTED_TENSOR_DTYPES = [
+    torch.bool,
+    torch.uint8,
+    torch.int8,
+    torch.int16,
+    torch.int32,
+    torch.int64,
+    # torch.float16,
+    torch.float32,
+    torch.float64,
+    # torch.complex32,
+    # torch.complex64,
+    # torch.complex128,
+    # torch.bfloat16,
+]
+
+
+class VariableType(Enum):
+    Bool = bool
+    Int = int
+    Float = float
+    String = str
+    ScalarDtype = ScalarDtype
+    TensorDtype = torch.dtype
+    Tuple = tuple
+
+    @staticmethod
+    def contains(v):
+        return v in [member.value for member in VariableType]
+
+
+def check_vtype(vtype, v):
+    if v is None:
+        return
+    if vtype in [bool, int, float]:
+        if type(v) not in [bool, int, float]:
+            raise TypeError("Variable type mismatch")
```

Comment:
Do we really need to raise here? Or we can return true/false.

Reply:
Yes, we can return true/false, but then I will have to repeat that line `raise TypeError("Variable type mismatch")` every time.

Reply:
Ok, updated the code. Now it returns boolean.